src/xmipp/libraries/reconstruction/tomo_filter_coordinates.cpp

```diff
@@ -0,0 +1,248 @@
+/***************************************************************************
+ *
+ * Authors:    Federico P. de Isidro-Gómez    fp.deisidro@cnb.csic.es (2021)
+ *
+ * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.csic.es'
+ ***************************************************************************/
+
+#include "tomo_filter_coordinates.h"
+#include <core/metadata_extension.h>
+#include <core/metadata_vec.h>
+#include <chrono>
+
+
+// --------------------------- INFO functions ----------------------------
+
+void ProgTomoFilterCoordinates::readParams()
+{
+	fnInTomo = getParam("--inTomo");
+	fnMask = getParam("--mask");
+	fnInCoord = getParam("--coordinates");
+	radius = getIntParam("--radius");
+    checkResThr = checkParam("--threshold");
+   	fnOut = getParam("-o");
+}
+
+
+void ProgTomoFilterCoordinates::defineParams()
+{
+	addUsageLine("This program filters a set of coordinates with one of two criteria:.");
+	addUsageLine("\nUsing a mask. In this case a set of coordinates and a mask with the regions to preserve ");
+    addUsageLine("coordinates different than 0 need to be input. If only these two options are input this ");
```

Comment:
can you revise this text? To me it's kinda confusing

Reply:
Done. Very reasonable.